syntax = "proto3";

package auth;

service AuthService {
  rpc Register (RegisterRequest) returns (AuthResponse);
  rpc Login (LoginRequest) returns (AuthResponse);
  rpc LoginWithGoogle (GoogleLoginRequest) returns (AuthResponse);
  rpc LoginWithTelegram (TelegramLoginRequest) returns (AuthResponse);
  rpc RefreshToken (RefreshTokenRequest) returns (AuthResponse);
  rpc RequestPasswordReset (RequestPasswordResetRequest) returns (RequestPasswordResetResponse);
  rpc ResetPassword (ResetPasswordRequest) returns (ResetPasswordResponse);
}

message RegisterRequest {
  string email = 1;
  string name = 2;
  string password = 3;
}

message LoginRequest {
  string email = 1;
  string password = 2;
}

message GoogleLoginRequest {
  string id_token = 1;
}

message TelegramLoginRequest {
  string telegram_id = 1;
  string auth_data = 2;
}

message AuthResponse {
  string access_token = 1;
  string refresh_token = 2;
}

message RefreshTokenRequest {
  string refresh_token = 1;
}

message RequestPasswordResetRequest {
  string email = 1;
}

message RequestPasswordResetResponse {
  bool success = 1;
}

message ResetPasswordRequest {
  string reset_token = 1;
  string new_password = 2;
}

message ResetPasswordResponse {
  bool success = 1;
}